client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -130,7 +130,8 @@
 
     // Generate random numbers for request ids.
     private final Supplier<Long> requestIdGenerator = RandomFactory.create()::nextLong;
-
+    
+    private final long deadline;
```

Comment:
You should call this timeout and add the unit of measure suffix (`timeoutMillis`). Deadline is not really a commonly used term.